tests/object.rs

```diff
@@ -76,4 +76,23 @@ compile_pass! {
         obj.drop();
         assert.equal(api.was_dropped(), true);
     ),
+    (
+    export class Api {
+        constructor();
+
+        static fetch(url, imports): Promise<void>;
+
+        create(value: number): CustomType;
+
+        was_dropped(): boolean;
+    }
+
+    export class CustomType {
+        constructor(api: Api, box: Box);
+
+        static new_(value: number): CustomType;
```

Comment:
missing api as first argument

Reply:
Hm? Where would that come from?

Reply:
so here `new_(api, value: number)` uses the wasm instance in api to create a Box. then the constructor is called with `new CustomType(api, box)`

Reply:
Sorry I don't get it. Where's the first argument `api` added?

Reply:
in `generate_function` there is an `#api_arg` which is added if it's `static`.